recipe/schemas/expression_grammar.py

```diff
@@ -1119,35 +1149,106 @@ def parse(
                 ColumnElement: A SQLALchemy expression
                 DataType: The datatype of the expression (bool, date, datetime, num, str)
         """
+        key = mkkey(
+            "parsed-ingredient",
+            text,
+            forbid_aggregation,
+            enforce_aggregation,
+            convert_dates_with,
+            convert_datetimes_with,
+        )
+        cache_result = (
+            self.cached_trees.get(key) if self.cached_trees is not None else None
+        )
+
+        def tree_to_expression(t, v):
+            return self.tree_to_expression(
+                t,
+                v,
+                key,
+                enforce_aggregation,
+                debug,
+                convert_dates_with,
+                convert_datetimes_with,
+            )
```

Comment:
This tree_to_expression wrapper seems confusing. Maybe this is better

```
extra_args = (key, enforce_aggragation, debug, convert_dates_with, convert_datetimes_with)
```

Then call

```
return tree_to_expression(tree, validator, *extra_args)
```

Reply:
:+1: